.config/tsconfig.workspace.json

```diff
@@ -5,6 +5,7 @@
     "allowSyntheticDefaultImports": true,
     "allowJs": true,
     "checkJs": true,
+    "skipLibCheck": false,
```

Comment:
👎 

To fix the problem which arose when enabling lib checks, this PR adds `@types/tmp` to our dependencies; this fixed an error caused by the author of `tmp-promise`.  Note that the invalid types which `@types/tmp` fixes are not referenced by our codebase; _we don't  gain any type fidelity_ by fixing this problem. Further, these types _are not exposed to our downstream consumers_--they aren't even exposed to _us_, really--so we cannot argue that we're doing this for the benefit of downstream.  And we _especially_ cannot argue this _because half of our codebase is still completely untyped_.

Depending upon types _for a transitive dependency_ seems dubious, yes?

It's a example of the hazard `skipLibCheck: false`; other people's problems will now be breaking our builds, and it will often be unclear whether the mitigation is a net-positive. A modest proposal: we should also run all of our dependencies' test suites and ensure they pass!!

If a downstream type is broken--and it actually _matters_--we will notice because we will be confronted with an unexpected `any` type.  We can then choose to stub the type out, or take action such as this (but with `skipLibCheck: false`, a downstream fix is the _only_ option; we have little agency).  Here, nobody (not us nor downstream) would have noticed a problem.  This poor tradeoff is the main reason why `skipLibCheck: true` is the default behavior.

Admittedly, the problems the lib check surfaced were few, but it's a mistake to think this will always be so. More problems of this nature will be surfaced the closer we get to complete type coverage. Some problems will not be so easily fixable short of using  `patch-package`.

That said, to ensure _we're_ shipping correct types--for the benefit of downstream--purpose-built tools such as [tsd](https://npm.im/tsd) or [expect-type](https://npm.im/expect-type) will prove a good investment. Even so, top priority should be type coverage. 

Also, `skipLibCheck: false` makes compilation way slower. 😜 

At the very least, I hope we can agree to defer a decision to enable lib checks until we're shipping complete type declarations for all of our packages.  I'll still argue vociferously against doing so, but I'm open to being swayed the other way once it doesn't seem so premature.